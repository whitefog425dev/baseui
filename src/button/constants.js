/* eslint-disable flowtype/require-valid-file-annotation */

export const KIND = {
  primary: 'primary',
  secondary: 'secondary',
  tertiary: 'tertiary',
  minimal: 'minimal',
};

export const SHAPE = {
  default: 'default',
  round: 'round',
  square: 'square',
};

export const SIZE = {
  default: 'default',
  compact: 'compact',
};
