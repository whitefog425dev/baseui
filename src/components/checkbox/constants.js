// @flow
/* eslint-disable import/prefer-default-export */

export const STATE_TYPE = {
  change: 'CHANGE',
};

/* eslint-enable import/prefer-default-export */
