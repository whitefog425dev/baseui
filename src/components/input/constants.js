// @flow
export const STATE_CHANGE_TYPE = {
  change: 'change',
};

export const ADJOINED = {
  none: 'none',
  left: 'left',
  right: 'right',
  both: 'both',
};

export const SIZE = {
  default: 'default',
  compact: 'compact',
};

export const ENHANCER_POSITION = {
  start: 'start',
  end: 'end',
};
