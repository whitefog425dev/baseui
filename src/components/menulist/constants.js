// @flow
/* eslint-disable import/prefer-default-export */

export const STATE_CHANGE_TYPES = {
  moveUp: 'moveUp',
  moveDown: 'moveDown',
};

// Dict of props to prepend $ to pass thru to styletron
export const STYLETRON_PROP_MAPPER = {
  ref: true,
  isHighlighted: true,
};

export const KEY_STRINGS = {
  ArrowUp: 'ArrowUp',
  ArrowDown: 'ArrowDown',
  Enter: 'Enter',
};
