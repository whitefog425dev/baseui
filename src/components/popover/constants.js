/*
MIT License

Copyright (c) 2018 Uber Technologies, Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
*/
// @flow

export const PLACEMENT = {
  auto: 'auto',
  topLeft: 'topLeft',
  top: 'top',
  topRight: 'topRight',
  rightTop: 'rightTop',
  right: 'right',
  rightBottom: 'rightBottom',
  bottomRight: 'bottomRight',
  bottom: 'bottom',
  bottomLeft: 'bottomLeft',
  leftBottom: 'leftBottom',
  left: 'left',
  leftTop: 'leftTop',
};

export const TRIGGER_TYPE = {
  click: 'click',
  hover: 'hover',
};

export const STATE_CHANGE_TYPE = {
  open: 'open',
  close: 'close',
};

export const ACCESSIBILITY_TYPE = {
  none: 'none',
  menu: 'menu',
  tooltip: 'tooltip',
};

export const POPOVER_MARGIN = 8;

export const ARROW_SIZE = 6;

/**
 * Since we use a 45-degree rotated div to render the arrow, the
 * width/height of this div is different than the arrow size itself
 *
 * The arrow size is essentially half the diagonal of the rotated div,
 * using pythagorean theorem:
 *   width^2 + height^2 = (arrow_size * 2)^2
 * In this case width = height so:
 *   2 * width^2 = (arrow_size * 2)^2
 * Simplifies to:
 *   width = âˆš((arrow_size * 2)^2 / 2)
 */
export const ARROW_WIDTH = Math.ceil(Math.sqrt((ARROW_SIZE * 2) ** 2 / 2));
