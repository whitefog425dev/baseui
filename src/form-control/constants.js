/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
// Dictionary of props to prepend $ to pass through to styletron
export const STYLETRON_PROP_MAPPER = {
  disabled: true,
  error: true,
  required: true,
  size: true,
};
