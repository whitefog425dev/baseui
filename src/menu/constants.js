/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const STATE_CHANGE_TYPES = {
  moveUp: 'moveUp',
  moveDown: 'moveDown',
};

// Dict of props to prepend $ to pass thru to styletron
export const SHARED_PROPS_MAPPER = {
  ref: true,
  isHighlighted: true,
};

export const KEY_STRINGS = {
  ArrowUp: 'ArrowUp',
  ArrowDown: 'ArrowDown',
  Enter: 'Enter',
  Space: ' ',
  Escape: 'Escape',
  Backspace: 'Backspace',
};

export const OPTION_LIST_SIZE = {
  default: 'default',
  compact: 'compact',
};
