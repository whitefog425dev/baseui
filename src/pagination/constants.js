// @flow

export const STATE_CHANGE_TYPE = {
  changePage: 'changePage',
};
