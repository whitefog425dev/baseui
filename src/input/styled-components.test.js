// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledLabel,
  StyledRoot,
  StyledInputEnhancer,
  StyledInputContainer,
  StyledInput,
  StyledCaption,
  SIZE,
  ADJOINED,
} from './index';
import {ENHANCER_POSITION} from './constants';

test('Input - StyledLabel - basic render', () => {
  const component = shallow(
    <StyledLabel $size={SIZE.default}>
      <span />
    </StyledLabel>,
  );
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledLabel has correct default styles',
  );
  component.setProps({
    $size: SIZE.compact,
    $disabled: true,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledLabel has correct styles when compact and disabled',
  );
});

test('Input - StyledRoot - basic render', () => {
  const component = shallow(
    <StyledRoot $size={SIZE.default}>
      <span />
    </StyledRoot>,
  );
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledRoot has correct default styles',
  );
  component.setProps({
    $size: SIZE.compact,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledRoot has correct styles when compact',
  );
});

test('Input - StyledInputEnhancer - basic render', () => {
  const component = shallow(
    <StyledInputEnhancer
      $size={SIZE.default}
      $position={ENHANCER_POSITION.start}
    >
      <span />
    </StyledInputEnhancer>,
  );
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputEnhancer has correct styles when default size and start position',
  );
  component.setProps({
    $size: SIZE.compact,
    $position: ENHANCER_POSITION.end,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputEnhancer has correct styles when compact and end position',
  );
});

test('Input - StyledInputContainer - basic render', () => {
  const component = shallow(
    <StyledInputContainer $size={SIZE.default} $adjoined={ADJOINED.none}>
      <span />
    </StyledInputContainer>,
  );
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputContainer has correct default styles',
  );
  component.setProps({
    $isFocused: true,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputContainer has correct styles when focused',
  );
  component.setProps({
    $size: SIZE.compact,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputContainer has correct styles when compact',
  );
  component.setProps({
    $adjoined: ADJOINED.left,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputContainer has correct styles when has start enhancer',
  );
  component.setProps({
    $adjoined: ADJOINED.right,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputContainer has correct styles when has end enhancer',
  );
  component.setProps({
    $adjoined: ADJOINED.both,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputContainer has correct styles when has start and end enhancers',
  );
  component.setProps({
    $error: true,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputContainer has correct styles when error',
  );
  component.setProps({
    $disabled: true,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInputContainer has correct styles when disabled',
  );
});

test('Input - StyledInput - basic render', () => {
  const component = shallow(<StyledInput $size={SIZE.default} />);
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInput has correct default styles',
  );
  component.setProps({
    $size: SIZE.compact,
    $disabled: true,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledInput has correct styles when compact and disabled',
  );
});

test('Input - StyledCaption - basic render', () => {
  const component = shallow(
    <StyledCaption $size={SIZE.default}>
      <span />
    </StyledCaption>,
  );
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledCaption has correct default styles',
  );
  component.setProps({
    $size: SIZE.compact,
    $error: true,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledCaption has correct styles when compact and error is boolean',
  );
  component.setProps({
    $error: <span>Error message</span>,
  });
  expect(component.instance().getStyles()).toMatchSnapshot(
    'StyledCaption has correct styles when error is a node',
  );
});
