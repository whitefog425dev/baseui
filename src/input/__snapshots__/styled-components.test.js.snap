// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input - StyledCaption - basic render: StyledCaption has correct default styles 1`] = `
Object {
  "color": "$theme.colors.mono800",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;

exports[`Input - StyledCaption - basic render: StyledCaption has correct styles when compact and error is boolean 1`] = `
Object {
  "color": "$theme.colors.mono800",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;

exports[`Input - StyledCaption - basic render: StyledCaption has correct styles when error is a node 1`] = `
Object {
  "color": "$theme.colors.negative400",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;

exports[`Input - StyledInput - basic render: StyledInput has correct default styles 1`] = `
Object {
  "::placeholder": Object {
    "color": "$theme.colors.mono700",
  },
  ":hover": Object {
    "cursor": "text",
  },
  "backgroundColor": "transparent",
  "borderStyle": "none",
  "borderWidth": "0",
  "boxSizing": "border-box",
  "caretColor": "$theme.colors.primary",
  "color": "$theme.colors.mono1000",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "outline": "none",
  "paddingBottom": "$theme.sizing.scale400",
  "paddingLeft": "$theme.sizing.scale500",
  "paddingRight": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale400",
  "width": "100%",
}
`;

exports[`Input - StyledInput - basic render: StyledInput has correct styles when compact and disabled 1`] = `
Object {
  "::placeholder": Object {
    "color": "$theme.colors.mono600",
  },
  ":hover": Object {
    "cursor": "not-allowed",
  },
  "backgroundColor": "transparent",
  "borderStyle": "none",
  "borderWidth": "0",
  "boxSizing": "border-box",
  "caretColor": "$theme.colors.primary",
  "color": "$theme.colors.mono600",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "outline": "none",
  "paddingBottom": "$theme.sizing.scale200",
  "paddingLeft": "$theme.sizing.scale500",
  "paddingRight": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale200",
  "width": "100%",
}
`;

exports[`Input - StyledInputContainer - basic render: StyledInputContainer has correct default styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono200",
  "borderColor": "$theme.colors.mono200",
  "borderRadius": "$theme.sizing.scale100",
  "borderStyle": "solid",
  "borderWidth": "1px",
  "boxShadow": "0 2px 6px transparent",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "border, boxShadow, backgroundColor",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "width": "100%",
}
`;

exports[`Input - StyledInputContainer - basic render: StyledInputContainer has correct styles when compact 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono100",
  "borderColor": "$theme.colors.primary400",
  "borderRadius": "$theme.sizing.scale100",
  "borderStyle": "solid",
  "borderWidth": "1px",
  "boxShadow": "0 2px 6px $theme.colors.shadowFocus",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "border, boxShadow, backgroundColor",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "width": "100%",
}
`;

exports[`Input - StyledInputContainer - basic render: StyledInputContainer has correct styles when disabled 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono300",
  "borderColor": "$theme.colors.mono300",
  "borderRadius": "0",
  "borderStyle": "solid",
  "borderWidth": "1px",
  "boxShadow": "0 2px 6px transparent",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono600",
  "display": "flex",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "border, boxShadow, backgroundColor",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "width": "100%",
}
`;

exports[`Input - StyledInputContainer - basic render: StyledInputContainer has correct styles when error 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono100",
  "borderColor": "$theme.colors.negative400",
  "borderRadius": "0",
  "borderStyle": "solid",
  "borderWidth": "1px",
  "boxShadow": "0 2px 6px $theme.colors.shadowError",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "border, boxShadow, backgroundColor",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "width": "100%",
}
`;

exports[`Input - StyledInputContainer - basic render: StyledInputContainer has correct styles when focused 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono100",
  "borderColor": "$theme.colors.primary400",
  "borderRadius": "$theme.sizing.scale100",
  "borderStyle": "solid",
  "borderWidth": "1px",
  "boxShadow": "0 2px 6px $theme.colors.shadowFocus",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "border, boxShadow, backgroundColor",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "width": "100%",
}
`;

exports[`Input - StyledInputContainer - basic render: StyledInputContainer has correct styles when has end enhancer 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono100",
  "borderColor": "$theme.colors.primary400",
  "borderRadius": "$theme.sizing.scale100 0 0 $theme.sizing.scale100",
  "borderStyle": "solid",
  "borderWidth": "1px",
  "boxShadow": "0 2px 6px $theme.colors.shadowFocus",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "border, boxShadow, backgroundColor",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "width": "100%",
}
`;

exports[`Input - StyledInputContainer - basic render: StyledInputContainer has correct styles when has start and end enhancers 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono100",
  "borderColor": "$theme.colors.primary400",
  "borderRadius": "0",
  "borderStyle": "solid",
  "borderWidth": "1px",
  "boxShadow": "0 2px 6px $theme.colors.shadowFocus",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "border, boxShadow, backgroundColor",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "width": "100%",
}
`;

exports[`Input - StyledInputContainer - basic render: StyledInputContainer has correct styles when has start enhancer 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono100",
  "borderColor": "$theme.colors.primary400",
  "borderRadius": "0 $theme.sizing.scale100 $theme.sizing.scale100 0",
  "borderStyle": "solid",
  "borderWidth": "1px",
  "boxShadow": "0 2px 6px $theme.colors.shadowFocus",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "border, boxShadow, backgroundColor",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "width": "100%",
}
`;

exports[`Input - StyledInputEnhancer - basic render: StyledInputEnhancer has correct styles when compact and end position 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono400",
  "borderRadius": "0 $theme.sizing.scale100 $theme.sizing.scale100 0",
  "color": "$theme.colors.mono900",
  "display": "flex",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "paddingBottom": "$theme.sizing.scale200",
  "paddingLeft": "$theme.sizing.scale500",
  "paddingRight": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale200",
}
`;

exports[`Input - StyledInputEnhancer - basic render: StyledInputEnhancer has correct styles when default size and start position 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono400",
  "borderRadius": "$theme.sizing.scale100 0 0 $theme.sizing.scale100",
  "color": "$theme.colors.mono900",
  "display": "flex",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "paddingBottom": "$theme.sizing.scale400",
  "paddingLeft": "$theme.sizing.scale500",
  "paddingRight": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale400",
}
`;

exports[`Input - StyledLabel - basic render: StyledLabel has correct default styles 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "display": "block",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": 500,
  "lineHeight": "$theme.typography.font350.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;

exports[`Input - StyledLabel - basic render: StyledLabel has correct styles when compact and disabled 1`] = `
Object {
  "color": "$theme.colors.mono700",
  "display": "block",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": 500,
  "lineHeight": "$theme.typography.font350.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;

exports[`Input - StyledRoot - basic render: StyledRoot has correct default styles 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "width": "100%",
}
`;

exports[`Input - StyledRoot - basic render: StyledRoot has correct styles when compact 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "width": "100%",
}
`;
